### ISSUE TITLE
semi rule doesn't check class properties

### ISSUE THREAD

**Issue by mikaello** [gajus/eslint-plugin-flowtype#210]
I seems like `eslint-plugin-flowtype` does not detect `props: <PropsObject>` or the state / defaultProps Flow annotations inside ES6 classes.

Excerpt from [React section in Flow-docs](https://flowtype.org/docs/react.html#defining-components-as-reactcomponent-subclasses):

```javascript
type Props = {
  title: string,
  visited: boolean,
  onClick: () => void,
};

class Button extends React.Component {
  props: Props;

  state: {
    display: 'static' | 'hover' | 'active',
  };

  static defaultProps: { visited: boolean };

  /* clipped */
}
```

.eslintrc
```javascript
// ...
"flowtype/semi": [2, "never"],
// ...
```

I get (correct) error on `type Props = { ... };`, _Extra semicolon. (flowtype/semi)_, but no errors on `props: Props;` or any of the other Flow annotations inside Button-class.

Is this intentional, or not part of what `eslint-plugin-flowtype` is supposed to cover?

**Comment by danharper** [comment]
Yep, this is a bug and not intentional.
